R/flipBox.R

```diff
@@ -0,0 +1,240 @@
+#' flipBox
+#' 
+#' Create a flipping box based off of the AdminLTE flipBox.
+#' 
+#' @param id the \code{flipBox} id
+#' @param front_content ui for the front of the flip box
+#' @param back_content ui for the back of the flip box
+#' @param front_text text for front button
+#' @param back_text text for back button
+#' 
+#' @section Warning: this function is experimental.
+#'
+#' @references \url{https://rinterface.github.io/shinydashboardPlus/reference/flipBox.html}
+#' 
+#' @return HTML tagList
+#' 
+#' @details To access the state of the flipbox use the input alias \code{input$<id>_value}.
+#'   For example, if your flipBox's id is myawesomeflipbox, you can access its state via
+#'   \code{input$myawesomeflipbox_value} where a 0 corresponds to the front, 1 the back, and NULL 
+#'   indicates the input has not been set yet (i.e. the card has not flipped yet). 
+#' 
+#' @examples
+#' library(shiny)
+#' library(shinydashboard)
+#' library(shinydashboardPlus)
+#' shinyApp(
+#'   ui = dashboardPage(
+#'     dashboardHeader(),
+#'     dashboardSidebar(),
+#'     dashboardBody(
+#'       flip_box(
+#'         id = "id",
+#'         front_content = div(
+#'           class = "text-center",
+#'           img(
+#'             src = "https://image.flaticon.com/icons/svg/149/149076.svg",
+#'             height = "300px",
+#'             width = "100%"
+#'           )
+#'         ),
+#'         back_content = div(
+#'           class = "text-center",
+#'           height = "300px",
+#'           width = "100%",
+#'           h1("Details...."),
+#'           p("More information....")
+#'         )
+#'       )
+#'     )
+#'   ),
+#'   
+#'   server = function(input, output, session) {
+#'   }
+#' )
+#'    
+#' @export
+#' @importFrom htmltools tagList
+#' @importFrom shiny singleton
+flipBox <- function(id, 
+                    front_content, 
+                    back_content, 
+                    front_text = "More Info",
+                    back_text = "Back to Front") {
+  
+  if (is.null(id) || missing(id)) stop("card id cannot be null or missing!")
+  
+  alias <- paste0(id, "_value")
```

Comment:
new "alias" to access state of flipBox (could maybe use "_state" suffix instead)